frontend/src/components/data-table/column-pinning/types.ts

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2024 Marimo. All rights reserved. */
```

Comment:
wait - there is actually already a column pinning feature: https://tanstack.com/table/v8/docs/api/features/column-pinning

so we can just re-use that instead of having our own logic

Reply:
there are two examples to do this:

* https://tanstack.com/table/v8/docs/framework/react/examples/column-pinning
* https://tanstack.com/table/v8/docs/framework/react/examples/column-pinning-sticky

it seems like you did something similar to the second (sticky and absolute positioning). i find the first may be a bit better / easier. 

you can do something like
```
const renderHeaderItem = (column) => {
   ...we have this logic
}

<THead>
 {table.getLeftHeaders().map(renderHeaderItem)}
 {table.getCenterHeaders().map(renderHeaderItem)}
 {table.getRightHeaders().map(renderHeaderItem)}
</THead>
```

thoughts?

Reply:
yeah this is way better, will change to this